### ISSUE TITLE
SSL exception when trying to proxy https

### ISSUE THREAD

**Issue by nileshprasadnz** [wiremock/wiremock#139]
Hi, Firstly thanks for making this tool available, it has really made my life easier. I am using it as a mock proxy, between my WebService client and an external webservice, to direct requests to my mock webservice implementation.  I can't get it to work using HTTPS.  I get the following exception.  Could you please advice if this is achievable using Wiremock.  Thanks a lot in advance. 

```
C:\workapps\wiremock>java -jar wiremock-1.46-standalone.jar --port 8081 --https-
port 8083 --https-keystore 
2014-05-14 07:33:42.288:INFO::Logging to STDERR via wiremock.org.mortbay.log.Std
ErrLog
2014-05-14 07:33:42.326:INFO::jetty-6.1.x
2014-05-14 07:33:42.336:INFO::Started DelayableSocketConnector@0.0.0.0:8081
2014-05-14 07:33:42.553:INFO::Started DelayableSslSocketConnector@0.0.0.0:8083
2014-05-14 07:33:48.159:WARN::EXCEPTION
javax.net.ssl.SSLException: Unrecognized SSL message, plaintext connection?
        at sun.security.ssl.InputRecord.handleUnknownRecord(Unknown Source)
        at sun.security.ssl.InputRecord.read(Unknown Source)
        at sun.security.ssl.SSLSocketImpl.readRecord(Unknown Source)
        at sun.security.ssl.SSLSocketImpl.performInitialHandshake(Unknown Source
)
        at sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
        at sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
        at wiremock.org.mortbay.jetty.security.SslSocketConnector$SslConnection.
run(SslSocketConnector.java:708)
        at com.github.tomakehurst.wiremock.jetty.DelayableSslSocketConnector$1.r
un(DelayableSslSocketConnector.java:54)
        at wiremock.org.mortbay.thread.QueuedThreadPool$PoolThread.run(QueuedThr
eadPool.java:582)
```

My setup is as follows:
Standalone WireMock using jre 1.7. Also tried 1.6.

```
java -jar wiremock-1.46-standalone.jar --port 8081 --https-port 8083
```

json mapping to direct to mocked webservice:

```
"request": {
            "method": "POST",
            "urlPattern": "/abc/webservice.*"
    },
    "response": {
            "proxyBaseUrl" : "http://localhost:8080/"
    } 
```

Client using apache cxf to invoke webservice:

``` java
          final String endpoint = "https://baseURL/abc/webservice";
        HTTPConduit http = (HTTPConduit) client.getConduit();
        http.getClient().setProxyServer("localhost");
        http.getClient().setProxyServerPort(8083);
```

Cheers,
Nilesh.

**Comment by darren-steven** [comment]
Is it feasible that an approach like charles-proxy could be uesd: Install a certificate chain on the client, and have the wiremock proxy do man in the middle by re-signing with it's own certificate on the fly?

**Comment by tomakehurst** [comment]
Do you mean for forward or reverse proxying?

**Comment by james-boswell** [comment]
@tomakehurst I appreciate this is an old post but a few post back, you state "Reverse proxing works because WireMock presents its own cert to the client."

I am trying to record HTTPS requests/responses using the proxy-all option. I believe this is known as reverse proxying. In this scenario, I get an SSLHandshakeException which I think would be because "WM presents its own cert to the client". By this, do you mean WM uses its own cert when communicating with the actual service it is recording requests/responses for? If so, can this behaviour be overridden?

**Comment by tomakehurst** [comment]
Since this isn't a bug report, please can you re-post this on the mailing list with your setup details as I suggested

**Comment by james-boswell** [comment]
@tomakehurst OK Tom, I will raise a new issue.

**Comment by tomakehurst** [comment]
Please can you post on the mailing list rather than opening an issue, thanks.

**Comment by james-boswell** [comment]
@tomakehurst I have raised issue #468. I will happily close it if you tell me how to access the mailing list?

**Comment by james-boswell** [comment]
@tomakehurst Closed issue #468 and raised through mailing list instead.
